### PR TITLE
Add Signal Icon

### PR DESCRIPTION
**Pull request by MiahaCybersec**
## What does this PR Change?

- Adds Signal Icon

While this does not address an existing issue, Signal recently added support for usernames, which will benefit those that use the Starlight theme and wish to publicly link their Signal profile. 

This is technically a duplicate of https://github.com/withastro/starlight/pull/1779, but I closed that pull request and created this one to comply with contributor guidelines.

### CODE REVIEW

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -103,6 +103,8 @@ export const BuiltInIcons = {
...
...19l1.09.27A10.9 10.9 0 0 1 2.69 6.374zm-.6 6.202a11 11 0 0 1 .122-1.63l-1.112-.168a12 12 0 0 0 0 3.596l1.112-.169A11 11 0 0 1 1.125 12zm17.078 10.275l-.578-.964a10.9 10.9 0 0 1-3.011 1.247l.27 1.091a12 12 0 0 0 3.319-1.374M22.875 12a11 11 0 0 1-.122 1.63l1.112.168a12 12 0 0 0 0-3.596l-1.112.169a11 11 0 0 1 .122 1.63zm.774 2.88l-1.09-.27a10.9 10.9 0 0 1-1.248 3.015l.964.581a12 12 0 0 0 1.374-3.326m-10.02 7.875a11 11 0 0 1-3.258 0l-.17 1.112a12 12 0 0 0 3.597 0zm7.125-4.303a11 11 0 0 1-2.304 2.302l.668.906a12 12 0 0 0 2.542-2.535zM18.45 3.245a11 11 0 0 1 2.304 2.304l.906-.675a12 12 0 0 0-2.535-2.535zM3.246 5.549A11 11 0 0 1 5.55 3.245l-.675-.906A12 12 0 0 0 2.34 4.874zm19.029.248l-.964.577a10.9 10.9 0 0 1 1.247 3.011l1.091-.27a12 12 0 0 0-1.374-3.318M10.371 1.246a11 11 0 0 1 3.258 0L13.8.134a12 12 0 0 0-3.597 0zM3.823 21.957L1.5 22.5l.542-2.323l-1.095-.257l-.542 2.323a1.125 1.125 0 0 0 1.352 1.352l2.321-.532zm-2.642-3.041l1.095.255l.375-1.61a10.8 10.8 0 0 1-1.21-2.952l-1.09.27a12 12 0 0 0 1.106 2.852zm5.25 2.437l-1.61.375l.255 1.095l1.185-.275a12 12 0 0 0 2.851 1.106l.27-1.091a10.8 10.8 0 0 1-2.943-1.217zM12 2.25a9.75 9.75 0 0 0-8.25 14.938l-.938 4l4-.938A9.75 9.75 0 1 0 12 2.25"/>',
```
Comment (HiDeoo) [inline packages/starlight/components/Icons.ts:107]:
```suggestion
		'<path d="m9.12.35.27 1.09a10.9 10.9 0 0 0-3.015 1.248l-.578-.964A12 12 0 0 1 9.12.35m5.76 0-.27 1.09a10.9 10.9 0 0 1 3.015 1.248l.581-.964A12 12 0 0 0 14.88.35M1.725 5.797A12 12 0 0 0 .351 9.119l1.09.27A10.9 10.9 0 0 1 2.69 6.374zm-.6 6.202a11 11 0 0 1 .122-1.63l-1.112-.168a12 12 0 0 0 0 3.596l1.112-.169A11 11 0 0 1 1.125 12zm17.078 10.275-.578-.964a10.9 10.9 0 0 1-3.011 1.247l.27 1.091a12 12 0 0 0 3.319-1.374M22.875 12a11 11 0 0 1-.122 1.63l1.112.168a12 12 0 0 0 0-3.596l-1.112.169a11 11 0 0 1 .122 1.63zm.774 2.88-1.09-.27a10.9 10.9 0 0 1-1.248 3.015l.964.581a12 12 0 0 0 1.374-3.326m-10.02 7.875a11 11 0 0 1-3.258 0l-.17 1.112a12 12 0 0 0 3.597 0zm7.125-4.303a11 11 0 0 1-2.304 2.302l.668.906a12 12 0 0 0 2.542-2.535zM18.45 3.245a11 11 0 0 1 2.304 2.304l.906-.675a12 12 0 0 0-2.535-2.535zM3.246 5.549A11 11 0 0 1 5.55 3.245l-.675-.906A12 12 0 0 0 2.34 4.874zm19.029.248-.964.577a10.9 10.9 0 0 1 1.247 3.011l1.091-.27a12 12 0 0 0-1.374-3.318M10.371 1.246a11 11 0 0 1 3.258 0L13.8.134a12 12 0 0 0-3.597 0zM3.823 21.957 1.5 22.5l.542-2.323-1.095-.257-.542 2.323a1.125 1.125 0 0 0 1.352 1.352l2.321-.532zm-2.642-3.041 1.095.255.375-1.61a10.8 10.8 0 0 1-1.21-2.952l-1.09.27a12 12 0 0 0 1.106 2.852zm5.25 2.437-1.61.375.255 1.095 1.185-.275a12 12 0 0 0 2.851 1.106l.27-1.091a10.8 10.8 0 0 1-2.943-1.217zM12 2.25a9.75 9.75 0 0 0-8.25 14.938l-.938 4 4-.938A9.75 9.75 0 1 0 12 2.25"/>',
```

It looks like we can compress a tiny bit the SVG path using [SVGOMG](https://jakearchibald.github.io/svgomg/).

File: .changeset/late-melons-ring.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Add Signal Icon
```
Comment (HiDeoo) [inline .changeset/late-melons-ring.md:5]:
```suggestion
Adds 1 new icon: `signal`
```

Tiny change as we always start with a present tense verb for changeset and for icons, we include the exact name users can use.

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for your contribution :raised_hands:

This is looking really great. I left a few suggestions but very minor ones. :thumbsup:

File: packages/starlight/schemas/social.ts
Comment (HiDeoo) [inline packages/starlight/schemas/social.ts:1]: Just noticed that a label is missing for the new entry (used by screen readers) but I cannot suggest a change for it as it's outside of the review range.

It's around line 71 and would look close to something like:

```diff
					openCollective: 'Open Collective',
+					signal: 'Signal',
```

Comment (MiahaCybersec) [reply packages/starlight/schemas/social.ts:1]: Good catch, I completely missed that. It should be added now.

**Review by HiDeoo — APPROVED** [review]
Thanks for the update, this is looking perfect to me 👍 

Note that this won't be merged until the next patch release, as this would update the Starlight documentation and we do not want to do that before the icon is actually available to users.

**Pull request merged**